### PR TITLE
Scheme helper fn

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

File: pkg/runtime/scheme/scheme_test.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	Describe("NewAddToScheme", func() {
+		It("should return a function that will add the types to the scheme", func() {
+			gv := schema.GroupVersion{Group: "core", Version: "v1"}
+			a := scheme.NewAddToScheme(gv, func() []runtime.Object {
+				return []runtime.Object{&v1.Pod{}, &v1.PodList{}}
+			})
+			s := runtime.NewScheme()
+
+			Expect(a(s)).To(Succeed())
+			Expect(s.AllKnownTypes()).To(HaveLen(13))
```
Comment (mengqiy) [inline pkg/runtime/scheme/scheme_test.go]: I don't understand why it has 13 items.
Maybe add a comments.

File: pkg/runtime/scheme/scheme_test.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+			Expect(s.AllKnownTypes()[gv.WithKind("Pod")]).To(Equal(reflect.TypeOf(v1.Pod{})))
+			Expect(s.AllKnownTypes()[gv.WithKind("PodList")]).To(BeEquivalentTo(reflect.TypeOf(v1.PodList{})))
```
Comment (mengqiy) [inline pkg/runtime/scheme/scheme_test.go]: Why this one is using `BeEquivalentTo` but not `Equal`?
What is the difference between Pod and PodList?

File: pkg/runtime/scheme/scheme.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// NewAddToScheme returns a function to add go struct types as GroupVersionKinds to a runtime.Scheme.
+// Scheme is used to map go structs to GroupVersionKinds and vice versa.
+func NewAddToScheme(gv schema.GroupVersion, types func() []runtime.Object) func(*runtime.Scheme) error {
```
Comment (DirectXMan12) [inline pkg/runtime/scheme/scheme.go]: This is a really weird name, and seems like a weird return type. Find a better name, maybe?

Comment (pwittrock) [reply pkg/runtime/scheme/scheme.go]: It creates a new `AddToScheme` function.

**Comment by pwittrock** [comment]
> I think the structure of this is fairly confusing. Why not have a helper like NewStandardScheme, which returns a new Scheme with these defaults already applied? You could still have this method available, but this seems like a weird entry point for common cases.

The approach of creating a NewAddToScheme function seems like a standard pattern across the kubernetes/kubernetes codebase.  This removes the need to copy-paste the code.  I think some of the code-generators rely upon the AddToScheme package function existing.

https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/abac/v1beta1/register.go

**Comment by mengqiy** [comment]
LGTM
golint is failing.

**Comment by pwittrock** [comment]
@mengqiy Fixed the comment :)

**Pull request merged**